metrics/qos/evm/metrics.go

```diff
@@ -49,13 +58,92 @@ var (
 	//   - End-to-end request success rates
 	//   - Error types by JSON-RPC method and chain
 	//   - HTTP status code distribution
+	//   - Service degradation when random endpoint fallback is used
 	requestsTotal = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Subsystem: pathProcess,
 			Name:      requestsTotalMetric,
 			Help:      "Total number of requests processed by EVM QoS instance(s)",
 		},
-		[]string{"chain_id", "service_id", "request_origin", "request_method", "success", "error_type", "http_status_code"},
+		[]string{"chain_id", "service_id", "request_origin", "request_method", "success", "error_type", "http_status_code", "random_endpoint_fallback"},
+	)
+
+	// availableEndpoints tracks the number of available endpoints per service.
+	// Labels:
+	//   - chain_id: Target EVM chain identifier
+	//   - service_id: Service ID of the EVM QoS instance
+	//
+	// Use to analyze:
+	//   - Endpoint pool size per service
+	//   - Service capacity trends
+	availableEndpoints = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: pathProcess,
+			Name:      availableEndpointsMetric,
+			Help:      "Number of available endpoints for EVM QoS instance(s)",
+		},
+		[]string{"chain_id", "service_id"},
+	)
+
+	// validEndpoints tracks the number of valid endpoints per service after filtering.
+	// Labels:
+	//   - chain_id: Target EVM chain identifier
+	//   - service_id: Service ID of the EVM QoS instance
+	//
+	// Use to analyze:
+	//   - Endpoint health per service
+	//   - Validation failure rates
+	//   - Service quality trends
+	validEndpoints = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: pathProcess,
+			Name:      validEndpointsMetric,
+			Help:      "Number of valid endpoints for EVM QoS instance(s) after validation",
+		},
+		[]string{"chain_id", "service_id"},
+	)
+
+	// endpointValidationsTotal tracks all endpoint validation attempts with detailed results.
+	// This metric provides comprehensive validation tracking for calculating success rates and analyzing failure patterns.
+	//
+	// Note: Multiple endpoint validations occur during each service request processing:
+	// - All available endpoints are validated before selection for the service request
+	// - Failed endpoints are filtered out with specific failure reasons captured
+	// - Valid endpoints are identified and one is selected to handle the request
+	// - This metric captures ALL validation attempts (both successful and failed) that occurred during endpoint selection
+	//
+	// Labels:
+	//   - chain_id: Target EVM chain identifier
+	//   - service_id: Service ID of the EVM QoS instance
+	//   - domain: eTLD+1 of endpoint URL for provider analysis (extracted from endpoint_addr)
+	//   - success: "true" for successful validations, "false" for failed validations
+	//   - validation_failure_reason: Specific failure reason for failed validations (empty for successful ones)
+	//
+	// Validation failure reasons include:
```

Comment:
Can you simply explain where this enum lives and what the prefix is? No one will keep it up to date.